src/views/Users.vue

```diff
@@ -1,28 +1,35 @@
 <template>
-  <user-list :users="users" />
+  <user-list />
 </template>
 <script>
+import { inject, onMounted } from "vue";
 import UserList from "@/components/UserList/UserList.vue";
-import { fetchUsers } from "@/services/HttpClientService.js";
-import { showNotification } from "@/utils/notification.js";
+
+import { useStore } from "vuex";
 
 export default {
   components: {
     UserList,
   },
-  data() {
-    return {
-      users: [],
-    };
-  },
-  async created() {
-      const response = await fetchUsers();
-      this.users = response;
+  setup() {
+    const store = useStore();
+    const showSnackbar = inject("showSnackbar");
+
+    onMounted(async () => {
+      try {
+        store.dispatch("users/fetchUsers");
+      } catch (error) {
+        showSnackbar({
+          message: error.message || "Failed to fetch users.",
```

Comment:
If I understand it correctly. we are always going to show the error.message if it's there. I think it's always going to be there. I think we should just show the "Failed to fetch users" message. I may be misunderstanding this, though.

Reply:
fixed